packages/devtools/test/support/cpu_profile_test_data.dart

```diff
@@ -0,0 +1,491 @@
+        }
+      ]);
+
+final subProfileTraceEvents = [
+  {
+    'ph': 'P',
+    'name': '',
+    'pid': 77616,
+    'tid': 42247,
+    'ts': 47377796685,
+    'cat': 'Dart',
+    'args': {'mode': 'basic'},
+    'sf': '140357727781376-5'
+  },
+  {
+    'ph': 'P',
+    'name': '',
+    'pid': 77616,
+    'tid': 42247,
+    'ts': 47377797975,
+    'cat': 'Dart',
+    'args': {'mode': 'basic'},
+    'sf': '140357727781376-8'
+  },
+  {
+    'ph': 'P',
+    'name': '',
+    'pid': 77616,
+    'tid': 42247,
+    'ts': 47377799063,
+    'cat': 'Dart',
+    'args': {'mode': 'basic'},
+    'sf': '140357727781376-11'
+  },
+];
+
+final responseWithMissingLeafFrame = {
+  'type': '_CpuProfileTimeline',
+  'samplePeriod': 1000,
+  'stackDepth': 128,
+  'sampleCount': 3,
+  'timeSpan': 0.003678,
+  'timeOriginMicros': 47377796685,
+  'timeExtentMicros': 3678,
+  'stackFrames': {
+    // Missing stack frame 140357727781376-0
+    '140357727781376-1': {
+      'category': 'Dart',
+      'name': 'thread_start',
+      'resolvedUrl': '',
+    },
+    '140357727781376-2': {
+      'category': 'Dart',
+      'name': '_pthread_start',
+      'parent': '140357727781376-1',
+      'resolvedUrl': '',
+    },
+    '140357727781376-3': {
+      'category': 'Dart',
+      'name': '_drawFrame',
+      'parent': '140357727781376-2',
+      'resolvedUrl': 'org-dartlang-sdk:///flutter/lib/ui/hooks.dart',
+    },
+    '140357727781376-4': {
+      'category': 'Dart',
+      'name': '_WidgetsFlutterBinding&BindingBase',
+      'parent': '140357727781376-3',
+      'resolvedUrl':
+          'file:///path/to/flutter/packages/flutter/lib/src/scheduler/binding.dart',
+    },
+  },
+  'traceEvents': [
+    {
+      'ph': 'P',
+      'name': '',
+      'pid': 77616,
+      'tid': 42247,
+      'ts': 47377796685,
+      'cat': 'Dart',
+      'args': {'mode': 'basic'},
+      'sf': '140357727781376-0'
+    },
+    {
+      'ph': 'P',
+      'name': '',
+      'pid': 77616,
+      'tid': 42247,
+      'ts': 47377797975,
+      'cat': 'Dart',
+      'args': {'mode': 'basic'},
+      'sf': '140357727781376-2'
+    },
+    {
+      'ph': 'P',
+      'name': '',
+      'pid': 77616,
+      'tid': 42247,
+      'ts': 47377799063,
+      'cat': 'Dart',
+      'args': {'mode': 'basic'},
+      'sf': '140357727781376-4'
+    },
+  ]
+};
+
+final TimeRange profileTime = TimeRange()
+  ..start = const Duration(microseconds: 0)
+  ..end = const Duration(microseconds: 100);
+final CpuStackFrame stackFrameA = CpuStackFrame(
+  id: 'id_0',
+  name: 'A',
+  category: 'Dart',
+  url: '',
+  profileTime: profileTime,
+  profileSampleCount: 10,
+)..exclusiveSampleCount = 0;
+final CpuStackFrame stackFrameB = CpuStackFrame(
+  id: 'id_1',
+  name: 'B',
+  category: 'Dart',
+  url: 'org-dartlang-sdk:///third_party/dart/sdk/lib/async/zone.dart',
+  profileTime: profileTime,
+  profileSampleCount: 10,
+)..exclusiveSampleCount = 0;
+final CpuStackFrame stackFrameC = CpuStackFrame(
```

Comment:
add line breaks between top level fields to improve readability.

Reply:
done